app/models/setting/cfgmgmt.rb

```diff
@@ -0,0 +1,29 @@
+class Setting::Cfgmgmt < Setting
+  def self.tool_specific
+    @tool_specific ||= {}
+  end
+
+  def self.add_tool_specific(tool_name, settings)
+    tool_specific[tool_name.to_sym] ||= []
+    tool_specific[tool_name.to_sym].concat(settings)
+  end
```

Comment:
Would it make sense to have this on the base setting (with another option for category) to enable easy extension of existing categories from plugins?

Reply:
I guess... But in that case, I'd prefer to finish the setting register and do it there.
As it is getting out of hand for this PR IMHO